src/main/java/com/google/codeu/servlets/Utils.java

```diff
@@ -0,0 +1,70 @@
+package com.google.codeu.servlets;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.appengine.datastore.AppEngineDataStoreFactory;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.DataStoreFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Collections;
+
+import javax.servlet.http.HttpServletRequest;
+
+
+class Utils {
+
+  /**
+   * Global instance of the {@link DataStoreFactory}. The best practice is to make it a single
+   * globally shared instance across your application.
+   */
+  private static final AppEngineDataStoreFactory DATA_STORE_FACTORY =
+          AppEngineDataStoreFactory.getDefaultInstance();
+
+  /** Global instance of the HTTP transport. */
+  static final HttpTransport HTTP_TRANSPORT = new UrlFetchTransport();
+
+  /** Global instance of the JSON factory. */
+  static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private static GoogleClientSecrets clientSecrets = null;
+
+  static GoogleClientSecrets getClientCredential() throws IOException {
+    if (clientSecrets == null) {
+      clientSecrets = GoogleClientSecrets.load(JSON_FACTORY,
+              new InputStreamReader(Utils.class.getResourceAsStream("/client_secrets.json")));
```

Comment:
Really nice! How did you deploy this file to appengine?

Oh.. I think I see that it's committed below to Github. This isn't ideal but I don't blame you for doing this. Let me check into this real quick to see if we can avoid this.

Reply:
I believe we can keep this out of the commit by setting environment variables instead? I can look into it when I have sec.

Reply:
That SGTM, but also due to time constraints, I'm also okay with this just getting submitted to GitHub. 😦 It is up to you.